api/src/main/java/org/jenkinsci/plugins/workflow/graph/FlowGraphWalker.java

```diff
@@ -13,8 +15,7 @@
  * @author Kohsuke Kawaguchi
  */
 // TODO guarantee DFS order (is it not already in DFS order?)
```

Comment:
Would be nice to actually test the order that is returned. For example, from

```groovy
stage '1'
parallel a: {
  stage '2'
  stage '3'
}, b: {
  node {
    stage '4'
  }
}
stage '5'
```

a DFS walk would I think produce the order 5, 3, 2, `parallel`, 1, 4, `node`, plus some non-`StepNode`s.